rn/Teacher/ios/Teacher/Submissions/QuizSubmissionList/Model/QuizSubmissionListInteractorLive.swift

```diff
@@ -0,0 +1,119 @@
+                quizID: String) {
+        self.context = Context(.course, id: courseID)
+        self.usersStore = env.subscribe(GetQuizSubmissionUsers(courseID: courseID))
+        self.submissionsStore = env.subscribe(GetAllQuizSubmissions(courseID: courseID, quizID: quizID))
+        self.quizStore = env.subscribe(GetQuiz(courseID: courseID, quizID: quizID))
+        self.courseStore = env.subscribe(GetCourse(courseID: courseID))
+
+        StoreState
+            .combineLatest(usersStore.statePublisher, submissionsStore.statePublisher, quizStore.statePublisher)
+            .subscribe(state)
+            .store(in: &subscriptions)
+
+        Publishers
+            .CombineLatest(usersStore.allObjects, submissionsStore.allObjects)
+            .map {
+                QuizSubmissionListItem.generateArray(users: $0.0, submissions: $0.1)
+            }
+            .combineLatest(filter) {
+                $0.applyFilter(filter: $1)
+            }
+            .subscribe(submissions)
+            .store(in: &subscriptions)
+
+        quizStore
+            .allObjects
+            .map {
+                $0.first?.title ?? ""
+            }
+            .subscribe(quizTitle)
+            .store(in: &subscriptions)
+
+        submissionsStore.exhaust()
+        usersStore.exhaust()
+        quizStore.refresh()
+        courseStore.refresh()
+    }
+
+    public func createMessageUserInfo() -> Future<[String: Any], Never> {
+        let quizTitle = quizTitle.value
+        let submissions = submissions.value
+        let contextCode = context.canvasContextID
+        let courseName = courseStore.first?.name ?? ""
+        return Future<[String: Any], Never> {  promise in
+            let recipients: [[String: Any?]] = submissions.map {
+                [
+                    "id": $0.id,
+                    "name": $0.displayName,
+                    "avatar_url": $0.avatarURL,
+                ] as [String: Any?]
+            }
+            let userInfo = [
+                "recipients": recipients,
+                "subject": quizTitle,
+                "contextName": courseName,
+                "contextCode": contextCode,
+                "canAddRecipients": false,
+                "onlySendIndividualMessages": true,
+            ]
+            promise(.success(userInfo))
+        }
+    }
+
+    public func setFilter(_ newFilter: QuizSubmissionListFilter) -> Future<Void, Never> {
+        Future<Void, Never> { [filter] promise in
+            filter.send(newFilter)
+            promise(.success(()))
+        }
+    }
+
+    // MARK: - Inputs
+    public func refresh() -> Future<Void, Never> {
+        Future<Void, Never> { [self] promise in
+            usersStore.exhaust(force: true)
+            quizStore.refresh(force: true)
+            self.submissionsStore.refreshWithFuture(force: true)
+                .sink { _ in promise(.success(())) }
+                .store(in: &self.subscriptions)
+        }
```

Comment:
I think we could just return the result of the `refreshWithFuture` method.
```suggestion
        usersStore.exhaust(force: true)
        quizStore.refresh(force: true)
        return submissionsStore.refreshWithFuture(force: true)
```